### ISSUE TITLE
Tab completion for repository path in `verdi computer setup` does not work

### ISSUE THREAD

**Issue by sphuber** [aiidateam/aiida_core#1830]
On the prompt for the working directory path, i.e. the repository, calling `<tab><tab>` will result in the prompt becoming unresponsive.

**Comment by broeder-j** [comment]
currently bash completion does not work at all on develop for me.

**Comment by sphuber** [comment]
We completely migrated `verdi` to the `click` infrastructure, so the old completion functionality no longer exists. Instead we rely on the completion that comes automatically with click. To activate it, put `eval "$(_VERDI_COMPLETE=source verdi)"` in your virtual env activation script and reload it

**Comment by broeder-j** [comment]
Thanks! I suspected that the command changed for click.

**Comment by sphuber** [comment]
This is a manifestation or duplicate of #1802

**Comment by giovannipizzi** [comment]
I'm not sure I can reproduce this (or if I understand this).
- If I type `verdi computer setup --work-dir ` and then I type `<TAB><TAB>`, completion works with the files I have in the current folder
- or do you mean pressing `<TAB>` at the interactive prompt? On my Mac, this simply does not work (meaning it goes a few chars ahead, but does not complete; if you then press backspace, I go back one only character). But it does not hang.

(I don't know if it's needed - I did a `pip install --upgrade .[dev_precommit]`)

What is instead the problem you see and the exact behaviour?

**Comment by sphuber** [comment]
The problematic behavior is indeed only for the interactive version. For any of the interactive options, at the prompt, typing `<TAB><TAB>` will simply insert a literal tab (or a few spaces in any case). It is annoying that pressing tab here is interpreted as a literal tab character and is not triggering the autocomplete. In addition, when you remove the tab, you can also remove part of the prompt label. And sometimes characters that you typed cannot be removed at all and will stay in the prompt. Example in `verdi computer setup` interactive, at the `workdir` option type a forward slash `/` followed by `<TAB>`. If you then press backspace multiple times, you remove the colon of the prompt label, but the slash that you typed remains. It seems like the prompt still works, as in you can type a valid value and it will get the right value, but it is just slightly annoying

**Comment by giovannipizzi** [comment]
I agree, but this has more to do with how click manages the interactive options: just ry in a script or in python
```python
click.prompt('a')
```
and you will see the same behaviour.

I don't know how much we can do with this, unless there is a 'click' way to fix this. I don't think this is critical, and actually if it's the shell is not blocked anymore, I would close this issue.

**Comment by giovannipizzi** [comment]
and actually, in the code, this boils down to a call to python's `raw_input` (or `input` in py3).

**Comment by giovannipizzi** [comment]
Apparently using deadline there are ways to tamper with raw_input
https://stackoverflow.com/questions/5637124/tab-completion-in-pythons-raw-input
I would anyway move this down in priority (and at least rename the issue)

**Comment by sphuber** [comment]
I agree that it is not critically blocking, but it wasn't marked as such. I would not close the issue but we can move priority to `nice-to-have` instead of important and leave this for another release.
